src/Magnum/Math/Random.h

```diff
@@ -0,0 +1,131 @@
+
+#include <random>
+#include <chrono>
+#include "Magnum/Types.h"
+#include "Magnum/Math/Constants.h"
+#include "Magnum/Math/Vector2.h"
+#include "Magnum/Math/Vector3.h"
+#include "Magnum/Math/Quaternion.h"
+#include "Magnum/Math/Functions.h"
+
+namespace Magnum
+{
+namespace Math
+{
+
+namespace Implementation
+{
+static std::seed_seq seeds{{
+    static_cast<std::uintmax_t>(std::random_device{}()),
+    static_cast<std::uintmax_t>(std::chrono::steady_clock::now()
+                                    .time_since_epoch()
+                                    .count()),
+}};
+
+class RandomGenerator
+{
+public:
+    RandomGenerator() = delete;
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Int, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_int_distribution<T>{start, end}(generator());
+    }
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Float, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_real_distribution<T>{start, end}(generator());
+    }
+
+public:
+    static std::mt19937 &generator()
+    {
+        static std::mt19937 g{seeds};
+        return g;
+    }
+};
+} // namespace Implementation
+namespace Random
+{
+template <class T = Float>
+T randomUnsignedScalar() // range [0, 1]
+{
+    return Implementation::RandomGenerator::generate<T>(static_cast<T>(0.0f),
+                                                        static_cast<T>(1.0f));
+}
+template <class T = Float>
+T randomSignedScalar() // range [-1, 1]
+{
+    return Implementation::RandomGenerator::generate(static_cast<T>(-1.0f), static_cast<T>(1.0f));
+}
+
+template <class T = Float>
+Vector2<T> randomUnitVector2()
+{
+    auto a = Implementation::RandomGenerator::generate(0.0f, 2 * Math::Constants<T>::pi());
+    return {std::cos(a), std::sin(a)};
+}
+
+template <class T = Float>
+Vector3<T> randomUnitVector3()
+{
+    auto a = Implementation::RandomGenerator::generate(0.0f, 2 * Math::Constants<T>::pi());
+    auto z = randomSignedScalar();
+    auto r = sqrt<T>(1 - z * z);
+    return {r * std::cos(a), r * std::sin(a), z};
+}
+template <class T = Float>
+Vector2<T> randomPointInACircle() // always length < 1
+{
+    while (true)
+    {
+        auto p = Vector2<T>(
+            randomSignedScalar(), randomSignedScalar());
+        if (p.length() >= 1)
+            continue;
+        return p;
+    }
+}
+
+template <class T = Float>
+Vector3<T> randomPointInASphere() // always length < 1
+{
+
+    while (true)
+    {
+        auto p = Vector3<T>(randomSignedScalar(),
+                            randomSignedScalar(),
+                            randomSignedScalar());
+        if (p.length() >= 1)
+            continue;
+        return p;
+    }
+}
```

Comment:
This has a non-deterministic runtime cost, which isn't great for realtime use -- I realized this could be doable as `randomUnitVector2()*randomUnsignedScalar()`, but then the user could do the same and so this function isn't really needed. So maybe remove both `randomPointInACircle()` and `randomPointInASphere()` and instead document that it can be achieved by combining `randomUnit*()` and `randomUnsignedScalar()`?

Reply:
This was my selfish implementation(not in your suggestions haha). I dont event think we need to document this; Get a unit sphere; multiply with a [0,1] length for example.